routing/index_router.cpp

```diff
@@ -254,11 +257,29 @@ IRouter::ResultCode IndexRouter::DoCalculateRoute(Checkpoints const & checkpoint
       return result;
 
     IndexGraphStarter::CheckValidRoute(subroute);
-    auto const nonFakefinish = IndexGraphStarter::GetNonFakeFinish(subroute);
-    startSegment = *nonFakefinish;
-    segments.insert(segments.end(), IndexGraphStarter::GetNonFakeStart(subroute), nonFakefinish);
+    auto const nonFakeStart = IndexGraphStarter::GetNonFakeStart(subroute);
```

Comment:
Is it possible to rename GetNonFakeStart() to GetNonFakeStartIt or GetNonFakeStartIterator? The same for GetNonFakeFinish. Also, I propose to mention somehow that nonFakeStart is an iterator actually.